src/backend/sgx/thread.rs

```diff
@@ -129,35 +141,106 @@ impl super::super::Thread for Thread {
-                                self.block.msg.req.arg[1].into(),
-                                self.block.msg.req.arg[2].into(),
-                                self.block.msg.req.arg[3].into(),
-                            )
-                            .map(|v| [v.into(), 0usize.into()])
-                            .map_err(|e| e.raw_os_error().unwrap_or(libc::EINVAL))
-                            .into()
-                        };
-
-                        return Ok(Command::Continue);
+                sallyport::host::execute(block.into_iter().filter_map(|item| {
+                    if cfg!(feature = "dbg")
+                        && !matches!(&item, Item::Syscall(syscall, ..) if syscall.num == 1)
+                        && !matches!(&item, Item::Gdbcall(..))
+                    {
+                        dbg!(&item);
                     }
 
-                    _ => return Ok(Command::SysCall(&mut self.block)),
-                }
+                    match item {
+                        #[cfg(feature = "gdb")]
+                        Item::Gdbcall(gdbcall, data) => {
+                            unsafe {
+                                execute_gdb(
+                                    gdbcall,
+                                    data,
+                                    &mut self.gdb_fd,
+                                    _gdblisten.as_ref().unwrap(),
+                                )
+                                .unwrap();
+                            }
+                            None
+                        }
+
+                        Item::Enarxcall(enarxcall, data) => match enarxcall {
+                            item::Enarxcall {
+                                num,
+                                argv: [leaf, subleaf, cpuid_offset, ..],
+                                ret,
+                            } if *num == item::enarxcall::Number::Cpuid => unsafe {
+                                let cpuid_ret =
+                                    deref_aligned::<CpuidResult>(data, *cpuid_offset, 1).unwrap();
+
+                                *cpuid_ret =
+                                    core::arch::x86_64::__cpuid_count(*leaf as _, *subleaf as _);
+                                *ret = 0;
+                                None
+                            },
+
+                            item::Enarxcall {
+                                num,
+                                argv: [target_info_offset, ..],
+                                ret,
+                            } if *num == item::enarxcall::Number::GetSgxTargetInfo => unsafe {
+                                let out_buf_ptr =
+                                    deref::<u8>(data, *target_info_offset, size_of::<TargetInfo>())
+                                        .unwrap();
+
+                                let out_buf =
+                                    slice::from_raw_parts_mut(out_buf_ptr, size_of::<TargetInfo>());
```

Comment:
Should we (possibly tech debt) add a `deref_slice` ?

Reply:
See https://github.com/enarx/sallyport/pull/108

Reply:
done